Initial load of features pages rescued from WayBackMachine

vicos59 opened the pull request:
Note: Just had to shorten the names of several folder and fnames because GitHub desktop had a fatal error because the path was too long. So, that is going to break a few things.

Most files have been cleaned up: removed old header/footer/menu and fixed links. Files in subdirs in features/features/ folder still need cleaned up.

Not integrated with main site yet. More work required.

Became more involved than I initially expected due to:

1. The number of files and folders involved;
2. The way WayBackMachine archives the files.

Not sure when I'll get back to this. So placing here to bus-proof it.

vicos59 committed:
Initial load of features pages rescued from WayBackMachine

Just had to shorten the names of several folder and fnames because GitHub desktop had a fatal error because the path was too long. So, that is going to break a few things.

Most files have been cleaned up: removed old header/footer/menu and fixed links. Files in subdirs in features/features/ folder still need cleaned up.

Not integrated with main site yet. More work required.

jonyo commented:
Wow, I didn't think too deeply about it but was not expecting this many different files just for the features page!  Trying to load the diff in github freezes the browser!

I'll check it out as I have time, maybe we can figure out how to consolidate the files to a single CSS file, the single HTML file for each page, and the images...  I know you've put a lot of work into it already to clean it up too, good work so far!

vicos59 commented:
haha, and this is after I already consolidated them into fewer folders. A lot of files...

I am probably going to rename features/features to features/details or something. I had to shorten it at the last minute because Github was barfing on the long pathnames.

Some files/folders may be missing in the features/features dir. I didn't get that far.

vicos59 commented:
oops, there are a few .bak files in there. I'll remove them from my copy.

jonyo commented:
OK As I'm going through it, going to make a list of things we'll need to remove.  I'll edit this comment to add more as I come across them.

Any links will be "localhost" links - they will only work if you have docker running for the website app and have this branch checked out.

A note on the "remove free" items - I don't think it would be good to keep the "free" wording as it implies the website sells some things, which it does not.  To be clear though, it would be good to make it clear the software is available free for download, but I think the mention of "free" on individual items might lead to confusion.  Maybe something at the top saying `All the features are part of the completely free GeoCore CE download` or similar.

- On the themes page: http://localhost:8080/features/geocore-software/geocore-themes.html
   - [ ] Remove The "FREE!" red banners on the graphics.
   - [ ] Maybe add wording saying the other themes can be downloaded from the resources page, and uploaded using the template set admin.  (or maybe link to the applicable page in the wiki)
   - [ ] I'm not 100% sure but I think the "attractive mobile viewing" at the bottom is the old iphone app, so remove that (and delete the pictures as you remove them from the page).
- [ ] Search for the word "free" in the text, and remove them (re-wording as needed).
- [ ] As you mentioned @vicos59 - remove the language pack stuff.
- [ ] Remove the `Mobile Integration` section and linked pages (if they are still there, the link seems 404) - iphone app is gone now.
- [ ] List of free addons on main page - remove: license activity, Joomla / Jfusion
- [ ] Addons page: http://localhost:8080/features/addons.html
   - [ ] Chat add-on - check on this one, just quickly skimming the code I don't see it.  And looking at the dedicated page http://localhost:8080/features/free-add-ons/597-live-chat-add-on.html it seems like maybe something Geo added to your site by hand, maybe with some JS code in your template or something.  In other words, not actually part of the software itself.
   - [ ] Remove joomla / jfusion addon - that had it's own repository that I don't have access to since they never needed me to work on it as a contractor.  Plus it is probably very out-dated and would need to be re-written (though that is likely the case for other integrations that ARE in the code... but still, a reason not to go out of my way to try to track it down).
   - [ ] Remove language packs from here as well.
   - [ ] Remove license activity log
   - [ ] Remove mobile API addon.

jonyo commented:
@vicos59 Given the amount of cleanup work needed...

I have an idea.  Instead of trying to delete stuff one by one, lets add stuff one by one.

What I can do is merge this to its own branch in the main repo, then both of us can be working on it.  And with it in its own branch, it won't show up on the main site until it is done.  We can do pull requests against the new branch instead of "main" until it is ready.

We can have a `legacy-features` folder (basically rename the folder you are adding now).  Then make a new `features` folder, we start off with an index.html, and copy in the markup part by part.  As we copy in an image, if we see it is still needed, copy that image over from the legacy folder, etc.

What is your thought on this approach @vicos59 ?

vicos59 commented:
Let me clean it up some more. I've been prudent about not copying files that are not needed. I do have to clean up the images folder. Better to have you focus on the first release.

I just wanted to get this in the repo in case I lost it, or got hit by a bus. Not ready for the web site yet.

vicos59 commented:
But yeah, another branch would be best.

jonyo commented:
Ah got it!

Yeah to be clear, the reason I suggested the "start from blank slate and add each part section by section" is to help with removing unused files, especially images.  But another way to go would be clean up all the files, then once we are done, I can find or make a script that looks at all the image / CSS / JS files and points out the ones it does not see any references to in any HTML files.  Maybe something like that would work too.

And that is fine too if you want to just let the PR sit and you can add to it over time.  What you can do is switch it to "draft" pull request - that makes it clear it is a work in progress.  Since it is already created - look in the top right, you'll see `Still in progress? Convert to Draft` - click on that.

vicos59 commented:
> I can find or make a script that looks at all the image / CSS / JS files and points out the ones it does not see any references to in any HTML files.

I used a different approach for js and css. I started with nada and used the network inspector to see which files were not loading and then added them in one-by-one. Since pages were generated by joomla, they all pretty much use the same stuff. There were a few cases of duplicates in diff dirs, so I consolidated into one folder.

There are very few css and js files actually in use, at least compared to how many I am used to seeing.

vicos59 committed:
Removed obsolete features

-- Removed not required images
-- Fixed page format on addon page
-- Removed obsolete features
-- Updated text to reflect new way of doing business in GeoCoreCE and provided links to new sites where applicable.

vicos59 committed:
Renamed features/features/ -> features/details/

vicos59 committed:
More cleanup on top level pages

-- Added back in the images deleted by mistake
-- moved themes page and cleaned up
-- Clean up on main features and addons pages.

vicos59 committed:
More cleanup & consolidation

-- Retrieved more missing images for addon pages. Still some missing and they do not even appear on the waybackmachine. Time to just remove them from the page.

-- Removed detailed features pages. They are redundant and offer no added value.

vicos59 committed:
Test Update

vicos59 committed:
Test II
